aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -576,4 +576,24 @@ static inline bool linalg_solve_is_vector_rhs(const Tensor& input, const Tensor&
   return vector_case;
 }
 
+static inline bool is_blas_compatible_column_major_order(const Tensor& input) {
+  IntArrayRef input_strides = input.strides();
+  IntArrayRef input_sizes = input.sizes();
+  auto ndim = input.dim();
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(ndim >= 2);
```

Comment:
I'm wondering in general why we often use ```ndim >= 2``` instead of ```ndim == 2```?

Reply:
Was setting the stage for the batched case and forgot to change it to `ndim == 2`.

Reply:
Does cuSPARSE do something special for the batched case, i.e. do they have specialized kernels here or are they flattening it internally and just dispatch to the 2d-Tensor kernels after some reshaping?

Reply:
It has specialized kernels for batched sparse matrix-dense matrix multiplications. From https://github.com/pytorch/pytorch/issues/60854#pytorch110-tasks:

> cuSPARSE supports batching Cᵢ = A ⋅ Bᵢ, Cᵢ = Aᵢ ⋅ B, Cᵢ = Aᵢ ⋅ Bᵢ.

Where A is a sparse matrix, B is a dense matrix, i is batch id.

The internal implementation of cuSPARSE is unknown, it's closed source.

Reply:
Ok we can ask NVIDIA or look at nvvp and check the trace of kernel launches on various input sizes

Reply:
Maybe cc @xwang233

Reply:
Batched operations are supported in cuSPARSE and may give better performance. We definitely want to support this use case in pytorch as well.